docs/user_guide/building_blocks/config_objects.md

```diff
@@ -53,9 +125,374 @@ given via the ```--variables``` argument, accessing information from columns bas
+# custom lfn retrieval method in case the underlying campaign is custom uhh
+if cfg.campaign.x("custom", {}).get("creator") == "uhh":
+    def get_dataset_lfns(
+        dataset_inst: od.Dataset,
+        shift_inst: od.Shift,
+        dataset_key: str,
+    ) -> list[str]:
+        # destructure dataset_key into parts and create the lfn base directory
+        dataset_id, full_campaign, tier = dataset_key.split("/")[1:]
+        main_campaign, sub_campaign = full_campaign.split("-", 1)
+        lfn_base = law.wlcg.WLCGDirectoryTarget(
+            f"/store/{dataset_inst.data_source}/{main_campaign}/{dataset_id}/{tier}/{sub_campaign}/0",
+            fs=f"wlcg_fs_{cfg.campaign.x.custom['name']}",
+        )
+
+        # loop though files and interpret paths as lfns
+        return [
+            lfn_base.child(basename, type="f").path
+            for basename in lfn_base.listdir(pattern="*.root")
+        ]
+
+    # define the lfn retrieval function
+    cfg.x.get_dataset_lfns = get_dataset_lfns
+
+    # define a custom sandbox
+    cfg.x.get_dataset_lfns_sandbox = dev_sandbox("bash::$CF_BASE/sandboxes/cf.sh")
+
+    # define custom remote fs's to look at
+    cfg.x.get_dataset_lfns_remote_fs = lambda dataset_inst: f"wlcg_fs_{cfg.campaign.x.custom['name']}"
+```
+
+### External_files
+
+If some files from outside columnflow are needed for an analysis, be them local files or online ( and accessible through wget), these can be indicated in the `cfg.x.external_files` auxiliary parameter.
+These can then be copied to the columnflow outputs using the {py:class}`~columnflow.tasks.external.BundleExternalFiles` task and used by being required by the object needing them.
+The `cfg.x.external_files` parameter expects a {py:class}`~columnflow.util.DotDict` with a string
+explaining the role of the file to be downloaded/copied as key and the link/path as value.
+It is also possible to give a tuple as value, with the link/path as the first entry of the tuple and a version as a second entry.
+As an example, the `cfg.x.external_files` parameter might look like this, for `json_mirror` the local path of the mirror directory to a specific commit of the [jsonPOG-integration Gitlab](https://gitlab.cern.ch/cms-nanoAOD/jsonpog-integration/-/tree/master) (CMS-specific):
+```python
+cfg.x.external_files = DotDict.wrap({
+    # lumi files
+    "lumi": {
+        "golden": ("/afs/cern.ch/cms/CAF/CMSCOMM/COMM_DQM/certification/Collisions17/13TeV/Legacy_2017/Cert_294927-306462_13TeV_UL2017_Collisions17_GoldenJSON.txt", "v1"),  # noqa
+        "normtag": ("/afs/cern.ch/user/l/lumipro/public/Normtags/normtag_PHYSICS.json", "v1"),
+    },
+
+    # muon scale factors
+    "muon_sf": (f"{json_mirror}/POG/MUO/2017_UL/muon_Z.json.gz", "v1"),
+
+    # some example of a ML training dataset to show an example with URL
+    "vehicules_dataset_fastai": ("https://github.com/arunoda/fastai-courses/releases/download/fastai-vehicles-dataset/fastai-vehicles.tgz", "v1"),  # noqa
+})
+```
```

Comment:
Import from example analysis?

Reply:
does not exist in this form, no example of an path with URL... I could do two examples if you really want to use the template here